src/Compilers/CSharp/Test/Symbol/Symbols/RequiredMembersTests.cs

```diff
@@ -0,0 +1,1052 @@
+            // (8,6): error CS9504: Do not use 'System.Runtime.CompilerServices.RequiredMemberAttribute'. Use the 'required' keyword on required fields and properties instead.
+            //     [RequiredMember]
+            Diagnostic(ErrorCode.ERR_ExplicitRequiredMember, "RequiredMember").WithLocation(8, 6),
+            // (9,16): warning CS0649: Field 'C.Field' is never assigned to, and will always have its default value 0
+            //     public int Field;
+            Diagnostic(ErrorCode.WRN_UnassignedInternalField, "Field").WithArguments("C.Field", "0").WithLocation(9, 16)
+        );
+
+        var prop = comp.SourceModule.GlobalNamespace.GetMember<PropertySymbol>("C.Prop");
+        Assert.False(prop.IsRequired);
+    }
+
+    [Fact]
+    public void UsingRequiredMemberAttributeExplicitly_WrongLocations()
+    {
+        var comp = CreateCompilation(@"
+using System;
+using System.Runtime.CompilerServices;
+class C
+{
+    [RequiredMember]
+    void M() {}
+    [RequiredMember]
+    event Action E;
+    [RequiredMember]
+    C() {}
+    [RequiredMember]
+    ~C() {}
+    [return: RequiredMember]
+    void M<[RequiredMember] T>([RequiredMember] int i) {}
+}
+
+namespace System.Runtime.CompilerServices
+{
+    public class RequiredMemberAttribute : Attribute
+    {
+        public RequiredMemberAttribute()
+        {
+        }
+    }
+}
+");
+
+        comp.VerifyDiagnostics(
+            // (9,18): warning CS0067: The event 'C.E' is never used
+            //     event Action E;
+            Diagnostic(ErrorCode.WRN_UnreferencedEvent, "E").WithArguments("C.E").WithLocation(9, 18)
+        );
+    }
+
+    [Fact]
+    public void RequiredWithInitializer()
+    {
+        var comp = CreateCompilationWithRequiredMembers(@"
+class C
+{
+    public required int Field = 1;
+    public required int Prop { get; set; } = 1;
+}
+");
+
+        // PROTOTYPE(req): Confirm with LDM whether we want a warning here.
+        comp.VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void RefReturningProperties()
+    {
+        var comp = CreateCompilationWithRequiredMembers(@"
+class C
+{
+    private int i;
+    public required ref int Prop => ref i;
+}
+");
+
+        // PROTOTYPE(req): Confirm with LDM whether we want an error here.
+        comp.VerifyDiagnostics(
+            // (5,29): error CS9505: Required member 'C.Prop' must be settable.
+            //     public required ref int Prop => ref i;
+            Diagnostic(ErrorCode.ERR_RequiredMemberMustBeSettable, "Prop").WithArguments("C.Prop").WithLocation(5, 29)
+        );
+    }
+
+    [Fact]
+    public void UnsettableMembers()
+    {
+        var comp = CreateCompilationWithRequiredMembers(@"
+#pragma warning disable CS0649 // Unassigned field
+class C
+{
+    public required readonly int Field;
+    public required int Prop1 { get; }
+}
+");
+
+        // PROTOTYPE(req): Confirm with LDM whether we want an error here.
```

Comment:
nit: The `readonly` question also extends to readonly properties and readonly structs.

Reply:
I do not agree.